packages/web/hooks/input/use-search-query-input.ts

```diff
@@ -8,20 +9,30 @@ import { useDebouncedState } from "../use-debounced-state";
  *  to perform a search remotely (is debounced). */
 export function useSearchQueryInput(debounceMs = 500) {
   // get selectable currencies for trading, including user balances if wallect connected
-  const [searchInput, setSearchInput_] = useState<string>("");
+  const [searchInput, setSearchInput_, isReady] = useQueryParamState<string>(
+    "s",
+    ""
+  );
 
   // generate debounced search from user inputs
   const [debouncedSearchInput, setDebouncedSearchInput] =
     useDebouncedState<string>("", debounceMs);
+
   const setSearchInput = useCallback(
     (input: string) => {
       const sanitizedSearchInput = input.replace(/#/g, "").trim();
       setSearchInput_(sanitizedSearchInput);
       setDebouncedSearchInput(sanitizedSearchInput);
     },
-    [setDebouncedSearchInput]
+    [setDebouncedSearchInput, setSearchInput_]
   );
 
+  useEffect(() => {
+    if (searchInput && isReady) {
+      setSearchInput(searchInput);
+    }
+  }, [isReady]);
```

Comment:
Is this effect needed? Is this initializing the state?

Reply:
I would rather not, maybe some refactoring should be done, the problem is that a state is kept inside `AllPoolsTable` which is called `searchQuery` which is an object, but the method that sets it is injected inside `TableControls`, if I don't use the `useEffect` when there is the first load of the page, the filter state passed to trpc is the empty one, as the one `searchQuery` is undefined, because nobody called `setSearchQuery`.

I didn't make a complete change because I don't know what other cases you need to have as a `searchQuery` object. 
It may make sense to make the `search` parameter that we pass to trpc a useMemo (I see that we are not currently using the limit field)

Reply:
If we can modify it that way, I can remove the effect, because the state would remain consistent